packages/app/src/routers/SynchronizationRouter/useSynchronizeFileUploader.ts

```diff
@@ -44,8 +45,14 @@ export const useSynchronizeFileUploader = ({
     openConflictResolutionModal,
   } = useSynchronizeFileConflictResolver(accessToken, iModelName);
 
+  const synchContext = useContext(SynchronizationContext);
+
   const uploadFiles = React.useCallback(
     async (fileList: FileList, onSuccess?: () => void) => {
+      if (!synchContext.isAuthorized) {
+        synchContext.login();
+        return;
```

Comment:
do you mean to return here? same question with the connections table

Reply:
yep. This is asynchronous, so we need to stop the process and the user must restart. Not a great UX, but improvements will be made in general there

Reply:
made the login synchronous and added a negative toast should the user close the login window without being authorized